packages/dds/tree/docs/Shared Tree Philosophy.md

```diff
@@ -0,0 +1,254 @@
+In a graph, data can have many parents, so they either need to not put constraints on the child (which makes it hard to have any application invariants which the schema language can't express),
+or all need to be accounted for (which doesn't scale to complex applications well).
+Therefor SharedTree goes with he tree approach, where constraints and invariants are handled at the subtree level;
+There are plans to support references to nodes allowing more graph like data, but these will not be able to easily put requirements on those subtrees like their parents can.
+
+---
+
+This subtree focused approach also helps optimizations like those mentioned in "Efficient Data Storage" to be effective and encapsulated.
+
+Some areas of the SharedTree design that currently do not deliver ideal subtree isolation are:
+
+-   Schema: A given schema identifier has the same meaning in the entire tree.
+    This means that currently embedding two different versions of a sub-tree application in different places with mismatched schema versions might not work depending on how it versions its schema.
+    Designs to address this, and some related issues (like generically parameterized collections) are referred to as "contextual schema" and could be implemented in future version of SharedTree.
+-   Loading and Summarization: Currently all subtrees are downloaded and summarized together.
+    Designs for "partial checkouts" are planned to address this for data loading, but there isn't a plan do address this for summarization.
+-   Op transmitting and processing: All clients will receive and process all ops in the shared tree.
+
+# Stored and View Schema
+
+Applications need to make assumptions about the data they are working with.
+For simple in-memory cases, this is generally managed with types.
+The code in the application uses those types to enable readers of the code (including other developers and compilers) to reason about and validate the logic in the application.
+Shared Tree provides this same experience: the application code gets to use types to reason about the document content it's working with, however it also has to deal with the fact that the document content might come from a different application or a different version of the same application (newer, older or even a divergent fork).
+To ensure content from a document is not misinterpreted, the document carries with it the "stored schema", which defines the structure and editing behaviors of the content within the document.
+This can be checked against the "view schema" which defines the types the application is programmed against.
+If they match, the application can safely work on the document as is.
+If they do not match, the application needs to take some corrective action.
+The set of possible corrective actions generally includes raising a compatibility error, updating or adapting the document to a supported format.
```

Comment:
```suggestion
The set of possible corrective actions generally includes raising a compatibility error, or updating or adapting the document to a supported format.
```
I think this helps clarify that it is the doc that is getting updated.

Reply:
The intention here is highlight the difference between the updating the schema chase and the case where the document content needs changes.
```suggestion
The set of possible corrective actions generally includes raising a compatibility error, updating the document's schema, or adapting the document's content to a supported format.
```